File: blaze_flutter_sdk/ios/blaze_flutter_sdk/Sources/Players/Tabs/BlazeReactPlayerTabsStyle.swift
//
//  BlazeReactPlayerTabsStyle.swift
//  blaze_flutter_sdk
//
//  Codable mirror of the Dart `BlazePlayerTabsStyle` (see
//  `moments_widget_tabs_configuration.dart`). Decoded with `toReactPlayerTabsStyle` and merged onto
//  the native `BlazePlayerTabsStyle.base()` — the same approach as the player-style customization,
//  so the dictionary is not parsed field-by-field.
//

import Foundation
import UIKit

struct BlazeReactPlayerTabsStyle: Codable {
    var padding: BlazeReactMargins?
    var gradient: BlazeReactPlayerTabsGradientStyle?
    var icon: BlazeReactPlayerTabItemIconStyle?
    var selectedTabState: BlazeReactPlayerTabItemStyle?
    var unselectedTabState: BlazeReactPlayerTabItemStyle?
    var activeTabIndicator: BlazeReactPlayerActiveTabIndicatorStyle?
    var isTabTitleVisibleWhenSingleTab: Bool?
    var isTabVisibleWhenEmpty: Bool?
}

struct BlazeReactPlayerActiveTabIndicatorStyle: Codable {
    var isVisible: Bool?
    var height: CGFloat?
}

struct BlazeReactPlayerTabsGradientStyle: Codable {
    var isVisible: Bool?
    var startColor: String? // Hex
    var middleColor: String? // Hex
    var endColor: String? // Hex
}

struct BlazeReactPlayerTabItemIconStyle: Codable {
    var iconTint: String? // Hex
    var padding: CGFloat?
}

struct BlazeReactPlayerTabItemStyle: Codable {
    var font: BlazeReactTitleFont?
    var textColor: String? // Hex
    var textSize: CGFloat?
    var letterSpacing: CGFloat?
    /// Android-only (API 29+). Ignored on iOS.
    var lineHeight: CGFloat?
}

File: blaze_flutter_sdk/android/src/main/kotlin/com/blaze/flutterblazesdk/players/tabs/BlazeReactPlayerTabsStyle.kt
package com.blaze.flutterblazesdk.players.tabs

import androidx.annotation.Keep
import com.blaze.flutterblazesdk.utils.parsing.BlazeReactTitleFont
import com.blaze.flutterblazesdk.widgets.customization.BlazeReactMargins

// JSON-serializable mirror of the Dart `BlazePlayerTabsStyle` (see
// `moments_widget_tabs_configuration.dart`). Deserialized with `toObject<...>()` and merged onto
// the native `BlazePlayerTabsStyle.base()` — same approach as the player-style customization, so we
// don't parse the creation-params map field-by-field.

@Keep
data class BlazeReactPlayerTabsStyle(
    val padding: BlazeReactMargins?,
    val gradient: BlazeReactPlayerTabsGradientStyle?,
    val icon: BlazeReactPlayerTabItemIconStyle?,
    val selectedTabState: BlazeReactPlayerTabItemStyle?,
    val unselectedTabState: BlazeReactPlayerTabItemStyle?,
    val activeTabIndicator: BlazeReactPlayerActiveTabIndicatorStyle?,
    val isTabTitleVisibleWhenSingleTab: Boolean?,
    val isTabVisibleWhenEmpty: Boolean?,
)

@Keep
data class BlazeReactPlayerActiveTabIndicatorStyle(
    val isVisible: Boolean?,
    val height: Float?,
)

@Keep
data class BlazeReactPlayerTabsGradientStyle(
    val isVisible: Boolean?,
    val startColor: String?, // Hex
    val middleColor: String?, // Hex
    val endColor: String?, // Hex
)

@Keep
data class BlazeReactPlayerTabItemIconStyle(
    val iconTint: String?, // Hex
    val padding: Float?,
)

@Keep
data class BlazeReactPlayerTabItemStyle(
    val font: BlazeReactTitleFont?,
    val textColor: String?, // Hex
    val textSize: Float?,
    val letterSpacing: Float?,
    val lineHeight: Float?,
)
